app/services/event_service.rb

```diff
@@ -4,6 +4,11 @@ def self.get_upcoming_events(user_id)
     parse_json(response)
   end
 
+  def self.get_past_events(user_id)
+    response = db_conn.get("/api/v1/users/#{user_id}/events?time_frame=past")
+    parse_json(response)
+  end
```

Comment:
We could refactor `get_upcoming_events` and `get_past_events` into a single service method (say `get_events`?) that takes the `time_frame` parameter?

Though I'm not sure this makes the code any more readable (maybe it just makes it more dynamic and reusable?), one potential solution:

```
def self.get_events(user_id, time_frame = nil)
  response = db_conn.get("/api/v1/users/#{user_id}/events") do |req|
               req.params['time_frame'] = time_frame if time_frame.present?
             end
  parse_json(response)
end
```

We would need to add some error handling on the back end to handle invalid `time_frame` params.  So that is one drawback with this approach.

Reply:
I will give this a stab - I think we can make this work pretty cleanly.  The BE only returns past events if the `time_frame` param is exactly equal to `past`, and will return upcoming events by default.  Not a lot of room for error, as this is all handled behind the scenes in our controller logic, and is abstracted away from user form input entirely.

Reply:
Love it!  Looks good to me!